File: go/model/education.go
package model

// EducationHistory represents a single education history entry.
type EducationHistory struct {
	StartYear   string `json:"startYear"`
	EndYear     string `json:"endYear"`
	School      string `json:"school"`
	Department  string `json:"department"`
	Description string `json:"description"`
}

File: haskell/src/Model/Education.hs
module Model.Education where

import Data.Aeson (ToJSON(..), object, (.=))
import Data.Text (Text)
import GHC.Generics (Generic)

data EducationHistory = EducationHistory
  { eduStartYear   :: Text
  , eduEndYear     :: Text
  , eduSchool      :: Text
  , eduDepartment  :: Text
  , eduDescription :: Text
  } deriving (Show, Generic)

instance ToJSON EducationHistory where
  toJSON e = object
    [ "startYear"   .= eduStartYear e
    , "endYear"     .= eduEndYear e
    , "school"      .= eduSchool e
    , "department"  .= eduDepartment e
    , "description" .= eduDescription e
    ]
